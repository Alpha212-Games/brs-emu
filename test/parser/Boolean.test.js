const Parser = require("../../lib/parser");
const Expr = require("../../lib/parser/Expression");
const { Lexeme } = require("../../lib/Lexeme");
const BrsError = require("../../lib/Error");

const { token, EOF } = require("./ParserTests");

describe("parser", () => {
    afterEach(() => BrsError.reset());

    describe("boolean expressions", () => {
        it("parses boolean ANDs", () => {
            let parsed = Parser.parse([
                token(Lexeme.True),
                token(Lexeme.And),
                token(Lexeme.False),
                EOF
            ]);

            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("parses boolean ORs", () => {
            let parsed = Parser.parse([
                token(Lexeme.True),
                token(Lexeme.Or),
                token(Lexeme.False),
                EOF
            ]);

            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });
});