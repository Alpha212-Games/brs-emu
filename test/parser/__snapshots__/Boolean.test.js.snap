// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser boolean expressions parses boolean ANDs 1`] = `
Array [
  Expression {
    "expression": Binary {
      "left": Literal {
        "value": BrsBoolean {
          "kind": 1,
          "value": true,
        },
      },
      "right": Literal {
        "value": BrsBoolean {
          "kind": 1,
          "value": false,
        },
      },
      "token": Object {
        "kind": 31,
        "line": 1,
        "literal": undefined,
      },
    },
  },
]
`;

exports[`parser boolean expressions parses boolean ORs 1`] = `
Array [
  Expression {
    "expression": Binary {
      "left": Literal {
        "value": BrsBoolean {
          "kind": 1,
          "value": true,
        },
      },
      "right": Literal {
        "value": BrsBoolean {
          "kind": 1,
          "value": false,
        },
      },
      "token": Object {
        "kind": 60,
        "line": 1,
        "literal": undefined,
      },
    },
  },
]
`;
